src/pyhf/__init__.py

```diff
@@ -82,6 +98,17 @@ def set_backend(backend, custom_optimizer=None):
                     backend.name, type(backend)
                 )
             )
+    if backend.precision not in _valid_precisions:
+        raise InvalidBackend(
```

Comment:
I thought I could still use `InvalidBackend` as the precision is a feature of the backend.

Reply:
I'd use `exceptions.Unsupported` instead.

Reply:
@kratsg Should be done now in the latest commit.